tests/ship_test.py

```diff
@@ -0,0 +1,235 @@
+
+        assert done, Print("ERROR: Did not find a \"done\" status for client %d" % (i))
+
+    Print("All clients active from block num: %s to block_num: %s." % (maxFirstBN, minLastBN))
+
+    stderrFile=Utils.getNodeDataDir(shipNodeNum, "stderr.txt")
+    biggestDeltaSeconds = 0.0
+    totalDeltaSeconds = 0.0
+    timeCount = 0
+    with open(stderrFile, 'r') as f:
+        line = f.readline()
+        while line:
+            match = re.search(r'info\s+([0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}.[0-9]{3})\s.+Received\sblock\s+.+\s#([0-9]+)\s@\s([0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}.[0-9]{3})',
+                              line)
+            if match:
+                rcvTimeStr = match.group(1)
+                prodTimeStr = match.group(3)
+                blockNum = int(match.group(2))
+                if blockNum > maxFirstBN:
+                    # ship requests can only affect time after clients started
+                    timeFmt = '%Y-%m-%dT%H:%M:%S.%f'
+                    rcvTime = datetime.strptime(rcvTimeStr, '%Y-%m-%dT%H:%M:%S.%f')
+                    prodTime = datetime.strptime(prodTimeStr, timeFmt)
+                    delta = rcvTime - prodTime
+                    deltaSeconds = delta.total_seconds()
+                    if deltaSeconds > biggestDeltaSeconds:
+                        biggestDeltaSeconds = deltaSeconds
+
+                    totalDeltaSeconds += deltaSeconds
+                    timeCount += 1
+                    assert deltaSeconds < 0.100, Print("ERROR: block_num: %s took more than %.3f seconds to be received." % (blockNum, deltaSeconds))
+
+            line = f.readline()
+
+    avg = totalDeltaSeconds / timeCount if timeCount > 0 else 0.0
+    Print("Greatest delay time: %s seconds, average delay time: %s seconds" % (biggestDeltaSeconds, avg))
+
+    testSuccessful = True
+finally:
+    TestHelper.shutdown(cluster, walletMgr, testSuccessful=testSuccessful, killEosInstances=killEosInstances, killWallet=killWallet, keepLogs=keepLogs, cleanRun=killAll, dumpErrorDetails=dumpErrorDetails)
+    if shipTempDir is not None:
+        if dumpErrorDetails and not testSuccessful:
+            for index in range(0, args.num_clients):
+                Print(Utils.FileDivider)
+                shipClientErrorFile = "%s%d.err" % (shipClientFilePrefix, i)
+                with open(shipClientErrorFile, "r") as f:
+                    Print("Contents of %s" % (shipClientErrorFile))
+                    line = f.readline()
+                    while line:
+                        Print(line)
+                        line = f.readline()
+                Print(Utils.FileDivider)
+                lineCount = 0
+                shipClientOutputFile = "%s%d.out" % (shipClientFilePrefix, i)
+                with open(shipClientOutputFile, "r") as f:
+                    Print("Contents of %s" % (shipClientOutputFile))
+                    line = f.readline()
```

Comment:
```
maxLines = 20000
for lineCount, line in zip(range(1, maxLines+1), f.readlines()):
    Print(line)
if lineCount == maxLines:
    Print("...       CONTENT TRUNCATED AT %d lines" % (maxLines))
```

Reply:
we discussed this and keeping the original done